### ISSUE TITLE
[Android] NullPointerException: Attempt to invoke virtual method 'double java.lang.Double.doubleValue()' on a null object reference

### ISSUE THREAD

**Issue by ivangr1** [software-mansion/react-native-reanimated#591]
Thanks for great library!
I am using `reanimated-bottom-sheet` in my app which depends on your library for animations, and I've been getting some crashes on Android lately and I traced them to your library.
I have two bottom sheets in my app and the crash happens when opening the second one but only for the first time. When I reload after crash and open it again it works fine.
I've tried all versions and the only solution that works for me is downgrading it to 1.4.0. It seems that later versions introduced the bug.

I am using React 16.9.0 and React Native 0.61.5

This is the whole stacktrace:
```
01-31 19:05:30.342 16685 16685 E unknown:ReactNative: java.lang.NullPointerException: Attempt to invoke virtual method 'double java.lang.Double.doubleValue()' on a null object reference
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.OperatorNode$21.eval(OperatorNode.java:186)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.OperatorNode$CompOperator.evaluate(OperatorNode.java:43)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.OperatorNode.evaluate(OperatorNode.java:298)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.NodesManager.getNodeValue(NodesManager.java:201)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.CondNode.evaluate(CondNode.java:22)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.NodesManager.getNodeValue(NodesManager.java:201)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.CondNode.evaluate(CondNode.java:25)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.OperatorNode$18.evaluate(OperatorNode.java:161)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.OperatorNode.evaluate(OperatorNode.java:298)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.NodesManager.getNodeValue(NodesManager.java:201)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.CondNode.evaluate(CondNode.java:22)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.BlockNode.evaluate(BlockNode.java:20)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.FunctionNode.evaluate(FunctionNode.java:18)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.CallFuncNode.evaluate(CallFuncNode.java:44)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.BlockNode.evaluate(BlockNode.java:20)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.NodesManager.getNodeValue(NodesManager.java:201)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.SetNode.evaluate(SetNode.java:22)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.BlockNode.evaluate(BlockNode.java:20)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.NodesManager.getNodeValue(NodesManager.java:201)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.CondNode.evaluate(CondNode.java:25)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.BlockNode.evaluate(BlockNode.java:20)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.NodesManager.getNodeValue(NodesManager.java:201)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.CondNode.evaluate(CondNode.java:25)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.BlockNode.evaluate(BlockNode.java:20)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.NodesManager.getNodeValue(NodesManager.java:201)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.TransformNode$AnimatedTransformConfig.getValue(TransformNode.java:27)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.TransformNode.evaluate(TransformNode.java:76)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.TransformNode.evaluate(TransformNode.java:14)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.StyleNode.evaluate(StyleNode.java:29)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.StyleNode.evaluate(StyleNode.java:14)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.PropsNode.evaluate(PropsNode.java:82)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.PropsNode.evaluate(PropsNode.java:19)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.value(Node.java:46)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.PropsNode.update(PropsNode.java:155)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.nodes.Node.runUpdates(Node.java:130)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.NodesManager.onAnimationFrame(NodesManager.java:160)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.NodesManager.access$000(NodesManager.java:55)
01-31 19:05:30.342 16685 16685 E unknown:ReactNative:   at com.swmansion.reanimated.NodesManager$1.doFr
```

**Comment by jakub-gonet** [comment]
Hey @ivangr1,
thanks for reporting.

Could you provide your code that triggers this crash? I'd like to see which part of the bottom sheet can cause this.

**Comment by Lucacel** [comment]
Hi I also encountered this issue any one has any solution ?

**Comment by draggie** [comment]
I can give example as Im also seeing this:
`<BottomSheet 
ref={this.otherOptionsRef}
          snapPoints={[0, 120]}
          initialSnap={0}
          enabledHeaderGestureInteraction={true}
          enabledContentGestureInteraction={false}
          onOpenEnd={this.onOpenHandler}
          onCloseEnd={this.onCloseEnd}
          renderHeader={this.renderHeader}
          renderContent={this.renderContent}
        />`
So quite a basic usage of the sheet, and then the ref is defined like:
`otherOptionsRef = React.createRef();`
Issue occurs after 
`  this.otherOptionsRef.current.snapTo(snapPoint);`
is being called twice (so for instance on the activation (showing) and then on the overlay click (hiding).

**Comment by draggie** [comment]
Ok problem is that this message is not very intuitive.
If you by accident have any snapTo called to, by accident like me, some valyue of snap not the position of snap, this will popup.
For example:
snapPoints={[0,120]}
and then if you call snapTo(120) the error will show after you try to snap from that position to something else.

**Comment by ivangr1** [comment]
> Hey @ivangr1,
> thanks for reporting.
> 
> Could you provide your code that triggers this crash? I'd like to see which part of the bottom sheet can cause this.

I decided to remove bottom sheet and replace it with static views after all. 
I think that the problem for me was not snapping like @draggie encountered.
I couldn't reproduce the error exactly. 
I tried all versions of reanimated and 1.6.0 crashes immediately after loading the bottom sheet, 1.7.0 crashes after first load of second sheet, and I thought that 1.4.0. doesn't crash but after a few days it also did crash with the same exception after a few openings and closings. 
Then I decided to refactor.
